src/storage-types/src/connections/aws.rs

```diff
@@ -136,58 +227,119 @@ impl AwsConfig {
             }
         };
 
-        let AwsCredentials {
-            access_key_id,
-            secret_access_key,
-            session_token,
-        } = &self.credentials;
-
-        let mut cred_provider = SharedCredentialsProvider::new(Credentials::from_keys(
-            access_key_id.get_string(secrets_reader).await.unwrap(),
-            secrets_reader
-                .read_string(*secret_access_key)
-                .await
-                .unwrap(),
-            match session_token {
-                Some(t) => Some(t.get_string(secrets_reader).await.unwrap()),
-                None => None,
-            },
-        ));
-
-        if let Some(AwsAssumeRole { arn }) = &self.role {
-            let mut role = AssumeRoleProvider::builder(arn).session_name("materialize");
-            // This affects which region to perform STS on, not where
-            // anything else happens.
-            if let Some(region) = &region {
-                role = role.region(region.clone());
+        let cred_provider = match &self.auth {
+            AwsAuth::Credentials(credentials) => {
+                let AwsCredentials {
+                    access_key_id,
+                    secret_access_key,
+                    session_token,
+                } = credentials;
+
+                SharedCredentialsProvider::new(Credentials::from_keys(
+                    access_key_id.get_string(secrets_reader).await.unwrap(),
```

Comment:
I know these `unwrap()`s were already here, but now that this is becoming production code they should turn into errors. They can say "internal error" in the message, because they _should_ never fail, but they can in rare cases fail (they involve a network call) and so we should be careful not to panic `envd` if they fail. A few more such instances below.